src/coreclr/jit/llvm.cpp

```diff
@@ -1643,47 +1405,298 @@ llvm::DILocation* createDebugFunctionAndDiLocation(struct DebugMetadata debugMet
+                    {
+                        // TODO-LLVM: out args?  E.g.
+                        //    /--*  t398   int    arg4 out+10
+                        //    +--*  t399   int    arg5 out + 14
+                        //    +--*  t908   byref  arg6 out + 18
+                        //    +--*  t397   int    arg3 in r9
+                        //    +--*  t395   ref    arg2 in r8
+                        //    +--*  t392   byref  arg0 in rcx
+                        //    +--*  t393   ref    arg1 in rdx
+                        failFunctionCompilation();
+                    }
+                    GenTreeCall::Use* oldArgs = callNode->gtCallLateArgs;
+                    callNode->ResetArgInfo();
+                    callNode->gtCallArgs = _compiler->gtPrependNewCallArg(returnAddrLcl, oldArgs);
+                    callNode->gtCallLateArgs = nullptr;
+                    _compiler->fgInitArgInfo(callNode);
+
+                    GenTree* returnAddrLcl2 = _compiler->gtNewLclvNode(returnTempNum, TYP_I_IMPL);
+                    GenTree* indirNode = _compiler->gtNewOperNode(callReturnType == var_types::TYP_STRUCT ? GT_OBJ : GT_IND, callReturnType, returnAddrLcl2);
+                    indirNode->gtFlags |= GTF_IND_TGT_NOT_HEAP; // No RhpAssignRef required
+                    LIR::Use callUse;
+                    if (CurrentRange().TryGetUse(callNode, &callUse))
+                    {
+                        callUse.ReplaceWith(_compiler, indirNode);
+                    }
+                    else
+                    {
+                        callNode->ClearUnusedValue();
+                    }
+
+                    callNode->gtReturnType = var_types::TYP_VOID;
+                    callNode->ChangeType(TYP_VOID);
+
+                    CurrentRange().InsertBefore(callNode, shadowStackVar);
+                    CurrentRange().InsertAfter(shadowStackVar, offset);
+                    CurrentRange().InsertAfter(offset, returnValueAddress);
+                    CurrentRange().InsertAfter(returnValueAddress, addrStore);
+                    CurrentRange().InsertAfter(addrStore, returnAddrLcl);
+                    CurrentRange().InsertAfter(callNode, returnAddrLcl2);
+                    CurrentRange().InsertAfter(returnAddrLcl2, indirNode);
+                }
+            }
+            else if (node->OperIs(GT_RETURN) && _functionSigHasReturnAddress)
+            {
+                var_types originalReturnType = node->TypeGet();
+                if(node->TypeIs(TYP_VOID))
+                {
+                    /* TODO-LLVM: retbuf .   compHasRetBuffArg doesn't seem to have an implementation */
+                    failFunctionCompilation();
+                }
+                //TODO-LLVM: how to create a GT_STORE_OBJ node?
+                if (originalReturnType == TYP_STRUCT)
+                {
+                    failFunctionCompilation();
+                }
+
+                assert(_retAddressLclNum == -1);
```

Comment:
```suggestion
                assert(_retAddressLclNum == BAD_VAR_NUM);
```
Or probably just have a step before the LLVM phases that initializes the special locals.